pkg/operator/starter.go

```diff
@@ -209,3 +220,40 @@ func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string)
 	}
 	return &ret.Status.OperatorStatusApplyConfiguration, nil
 }
+
+func withCustomEndPoint(infraLister v1.InfrastructureLister) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		infra, err := infraLister.Get(infrastructureName)
+		if err != nil {
+			return err
+		}
+		if infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.PowerVS == nil {
+			return nil
+		}
+		serviceEndPoints := infra.Status.PlatformStatus.PowerVS.ServiceEndpoints
+		if len(serviceEndPoints) == 0 {
+			return nil
+		}
+		var containerEnvVars []corev1.EnvVar
+		for _, serviceEndPoint := range serviceEndPoints {
+			if _, ok := endPointKeyToEnvNameMap[serviceEndPoint.Name]; !ok {
+				klog.Infof("Ignoring %q serviceEndpoint", serviceEndPoint.Name)
+				continue
+			}
+			containerEnvVars = append(containerEnvVars, corev1.EnvVar{
+				Name:  endPointKeyToEnvNameMap[serviceEndPoint.Name],
+				Value: serviceEndPoint.URL,
+			})
+		}
+
+		for i := range deployment.Spec.Template.Spec.Containers {
+			container := &deployment.Spec.Template.Spec.Containers[i]
+			if container.Name != "csi-driver" {
```

Comment:
See if required this can be changed into constant.

Reply:
Addressed!